### ISSUE TITLE
Breaks `$page->render()`

### ISSUE THREAD

**Issue by adamkiss** [getkirby/layouts#7]
I've been preparing an error issue for the [Kirby Static Site Generator](https://getkirby.com/plugins/d4l-data4life/static-site-generator) not working if layouts are used, but looking closely at the problem, it seems there's something weird happening with the `render()` function when the layouts plugin is used.

If I call `$page('error')->render()` with the layouts plugin on, the `$page` refers to the error (correct) page in the layout, but refers to the last rendered / first requested page (see edit) in the template itself.

You can see the problem in the kirby/ssg/layouts demo repo build files:

- layout: https://github.com/adamkiss/kirby-ssg-layouts-problem/blob/main/site/layouts/default.php
- template: https://github.com/adamkiss/kirby-ssg-layouts-problem/blob/main/site/templates/default.php
- built page: https://github.com/adamkiss/kirby-ssg-layouts-problem/blob/main/__build/error/index.html

result of `page('error')->render()`:

```html
<!DOCTYPE html>
<html class="no-js">

<head>
	<meta charset="UTF-8">
	<meta name="viewport" content="width=device-width">

	<title>Error • Site Title</title>
	<script type="module">
		document.documentElement.classList.remove('no-js');
		document.documentElement.classList.add('js');
	</script>

	<link rel="icon" href="/favicon.png">
	<link rel="icon" href="/favicon.svg" type="image/svg+xml">
</head>

<body class="antialiased dark:bg-gray-900 dark:text-white">

<h1>Home</h1>

</body>
</html>
```

Edit:

Order matters. templates get whatever page called 'render' first.

```php
// generates `<h1>Error</h1>` in the template body
page('error')->render();

// generates `<h1>Home</h1>` in the template body of both
page('home')->render(); page('error')->render();

// generates `<h1>Error</h1>` in the template body of both
page('error')->render(); page('home')->render();


```

**Comment by lukasbestle** [comment]
I believe it could be related to this line: https://github.com/getkirby/layouts/blob/29b743ed5cb047608566b86386acf2c22610fe8a/src/Layout.php#L41
Because it alters global state, it may break subsequent renders.

**Comment by adamkiss** [comment]
@lukasbestle layout variables work, whil the template variables stick from the last render. I can't visualise the flow properly, so I can't tell if it is that, or not.

**Comment by bastianallgeier** [comment]
This is solved in 3.9
